stl/inc/atomic

```diff
@@ -480,13 +490,37 @@ struct _Atomic_storage<_Ty, 1> { // lock-free using 1-byte intrinsics
 
     bool compare_exchange_strong(_Ty& _Expected, const _Ty _Desired,
         const memory_order _Order = memory_order_seq_cst) noexcept { // CAS with given memory order
-        const char _Expected_bytes = _Atomic_reinterpret_as<char>(_Expected); // read before atomic operation
+        char _Expected_bytes = _Atomic_reinterpret_as<char>(_Expected); // read before atomic operation
         char _Prev_bytes;
-        _ATOMIC_CHOOSE_INTRINSIC(_Order, _Prev_bytes, _InterlockedCompareExchange8, _Atomic_address_as<char>(_Storage),
-            _Atomic_reinterpret_as<char>(_Desired), _Expected_bytes);
-        if (_Prev_bytes == _Expected_bytes) {
-            return true;
+
+#if _HAS_CXX20
+        if constexpr (_STD has_unique_object_representations_v<_Ty>) {
+#endif
+            _ATOMIC_CHOOSE_INTRINSIC(_Order, _Prev_bytes, _InterlockedCompareExchange8,
+                _Atomic_address_as<char>(_Storage), _Atomic_reinterpret_as<char>(_Desired), _Expected_bytes);
+            if (_Prev_bytes == _Expected_bytes) {
+                return true;
+            }
+#if _HAS_CXX20
+        } else {
+            _Ty _Mask;
+            _CSTD memset(&_Mask, 0xff, sizeof(_Ty));
+            __builtin_zero_non_value_bits(_STD addressof(_Mask));

```

Comment:
In the tests, I see `clang-cl` failing with:

```
D:\build\x64\out\inc\atomic(508,13): error: use of undeclared identifier '__builtin_zero_non_value_bits'
            __builtin_zero_non_value_bits(_STD addressof(_Mask));
            ^
```

I assume this means that Clang 10 doesn't support this intrinsic. Is there a bug/feature request for them?

Reply:
For now I just changed preprocessor check. Will request and add if the change is otherwise fine.

Reply:
Actually I think I'll go ahead and create LLVM issue, if it isn't there, and add `TRANSITION` comment. But first will need to make the tests pass and the coverage complete.